test/integration/daemonset/daemonset_test.go

```diff
@@ -421,7 +424,10 @@ func forEachStrategy(t *testing.T, tf func(t *testing.T, strategy *apps.DaemonSe
 
 func TestOneNodeDaemonLaunchesPod(t *testing.T) {
 	forEachStrategy(t, func(t *testing.T, strategy *apps.DaemonSetUpdateStrategy) {
-		ctx, closeFn, dc, informers, clientset := setup(t)
+		_, ctx := ktesting.NewTestContext(t)
+		ctx, cancel := context.WithCancel(ctx)
+		defer cancel()
```

Comment:
`context.WithCancel` is not needed? Please check and remove it and also in the other tests.